pkg/controller/manager/manager_controller.go

```diff
@@ -743,3 +750,61 @@ func fillDefaults(mc *operatorv1.ManagementCluster) {
 		mc.Spec.TLS.SecretName = render.VoltronTunnelSecretName
 	}
 }
+
+func getVoltronRouteConfig(ctx context.Context, cli client.Client, managerNamespace string) (*render.VoltronRouteConfig, error) {
+	terminatedRouteList := &operatorv1.TLSTerminatedRouteList{}
+	if err := cli.List(ctx, terminatedRouteList); err != nil {
+		return nil, err
+	}
+
+	passThroughRouteList := &operatorv1.TLSPassThroughRouteList{}
+	if err := cli.List(ctx, passThroughRouteList); err != nil {
+		return nil, err
+	}
+
+	if len(terminatedRouteList.Items) == 0 && len(passThroughRouteList.Items) == 0 {
+		return nil, nil
+	}
+
+	builder := render.NewVoltronRouteConfigBuilder()
+	for _, route := range terminatedRouteList.Items {
+		if route.Spec.CABundle != nil {
```

Comment:
Just curious if you considered embedding all of this secret fetching logic within `AddTLSTerminatedRoute` and if so, why you opted to split it out here instead?

Reply:
I suppose we do have a pattern of not putting client code that interacts with the API in the render package. I think we'd want to move this out of the render package if we did that, which perhaps isn't the worst thing in the world and might alleviate your concerns about package structure from the other comment. I'm OK either way.